### PR TITLE
Update clap to 4.3

### PR DESCRIPTION
**Pull request by ljedrz**
This is the snarkOS counterpart to https://github.com/AleoHQ/snarkVM/pull/1583.

### CODE REVIEW

**Review by joske — APPROVED** [review]

**Comment by ljedrz** [comment]
Updated with 3 test cases for arg parsing.

File: cli/src/commands/mod.rs [hunk 1]
```diff
@@ -78,4 +86,99 @@ mod tests {
         use clap::CommandFactory;
         CLI::command().debug_assert()
     }
+
+    #[test]
+    fn clap_snarkos_start() {
```
Comment (howardwu) [inline cli/src/commands/mod.rs]: Move these to their respective files so their fields don't need to be `pub` exposed

Comment (ljedrz) [reply cli/src/commands/mod.rs]: They would still need to be `pub` there, as the modules would be separate (the `tests` module vs. the parent module); however, I don't see a reason why those fields wouldn't all be `pub` (like in `struct Start`), as they are only used to parse commands, so there is not much reason to not expose them.

Comment (howardwu) [reply cli/src/commands/mod.rs]: In my experience, they won't need `pub` if they're in the same file.

Comment (ljedrz) [reply cli/src/commands/mod.rs]: You're right, I must have mis-pasted something last time, which gave me import errors; updated and rebased :+1:.

**Review by howardwu — APPROVED** [review]
LGTM

**Pull request merged**